### PR TITLE
Bump antlr and urwid, fix resulting errors.

### PR DESCRIPTION
**Pull request by jcaesar**
Hi, I was trying to use pyfx as packaged in NixOS/nixpkgs. There, it's only packaged as a dependency of a different package, and invoking it directly fails with a relatively [ugly error](https://github.com/NixOS/nixpkgs/issues/333173) due to nixpkgs ignoring requirements.txt and using newer versions for antlr and urwid (python dependency hell problems…). I'm [trying to fix](https://github.com/NixOS/nixpkgs/pull/333880) that, I think the necessary patches would be worth having here, too.

### CODE REVIEW

File: src/pyfx/model/autocomplete/autocomplete_listener.py [hunk 1]
```diff
@@ -5,7 +5,10 @@
 from loguru import logger
 from overrides import overrides
 
-from ..common.jsonpath import JSONPathLexer, JSONPathParser, JSONPathListener
+from ..common.jsonpath.JSONPathLexer import JSONPathLexer
+from ..common.jsonpath.JSONPathParser import JSONPathParser
+from ..common.jsonpath.JSONPathListener import JSONPathListener
+
```
Comment (cielong) [inline src/pyfx/model/autocomplete/autocomplete_listener.py:11]: Nit: Please remove this line, it fails the style check. See https://github.com/cielong/pyfx/pull/135

**Review by cielong — APPROVED** [review]
Thanks so much for the change! Leave a minor comment to pass the CI/CD

File: src/pyfx/view/common/frame.py [hunk 2]
```diff
@@ -278,12 +278,12 @@ def get_cursor_coords(self, size):
...
 
     @overrides
-    def selectable(self):
+    def selectable(self) -> bool:
         """Frame is always selectable."""
         return True
 
     @overrides
-    def sizing(self):
+    def sizing(self) -> frozenset[urwid.Sizing]:
```
Comment (cielong) [inline src/pyfx/view/common/frame.py:286]: Looks like this is not supported in Py38, can you also remove it? https://stackoverflow.com/questions/75202610/typeerror-type-object-is-not-subscriptable-python

Comment (cielong) [reply src/pyfx/view/common/frame.py:286]: Or we can remove support for Py38, since it is an old version.

Comment (jcaesar) [reply src/pyfx/view/common/frame.py:286]: What do you think about the solution suggested in the SO you linked? i.e. `def sizing(self) -> typing.FrozenSet[urwid.Sizing]:`?

> can you also remove it?

I could remove the `@overrides`, if that's what you mean?

> remove support for Py38

That seems avoidable.

**Comment by cielong** [comment]
Thanks!

**Comment by jcaesar** [comment]
Thank you, for the cleanup and release too.

**Pull request merged**